dashboard/src/shared/types.ts

```diff
@@ -323,3 +337,13 @@ export interface IAppOverview {
   status: string;
   chart: string;
 }
+
+export interface IKubeItem {
```

Comment:
```suggestion
export interface IKubeItem<T> {
```

Reply:
We can make this a generic type so that we can define specifically what we're looking for whenever we use it. That way we can be explicit about when we only expect there to be `ISecret` items, and in other places we can be more generic and just have `IResource`. The current way this is done is less desirable as we'd have to enumerate each type we want to potentially support. What do you think?

Reply:
The problem with that is when using it in `IKubeState` we will find trouble when using `IKubeState.items` to something different than `IResource`. For the moment I can avoid the issues with that but I am not sure if that will be always the case.

Reply:
Hmm, what if IKubeState was also a generic and took a type param?

Reply:
yes, but that that won't work if the caller wants to use two different types of resources. In any case, that's not needed right now, let's evaluate that again if we find an issue.

Reply:
@andresmgot if the caller wants to use two different types of resources they can pass a union in, e.g. `IKubeState<IResource  | ISecret>` right? But yes, I agree we can reevaluate when we need it.